src/apps/chifra/internal/abis/output.go

```diff
@@ -20,18 +21,23 @@ import (
 // RunAbis handles the abis command for the command line. Returns error only as per cobra.
 func RunAbis(cmd *cobra.Command, args []string) (err error) {
 	opts := abisFinishParse(args)
+	outputHelpers.InitJsonWriter("abis", &opts.Globals)
 	// EXISTING_CODE
 	// EXISTING_CODE
 	err, _ = opts.AbisInternal()
+	outputHelpers.CloseJsonWriterIfNeeded("abis", &opts.Globals)
 	return
 }
 
 // ServeAbis handles the abis command for the API. Returns error and a bool if handled
 func ServeAbis(w http.ResponseWriter, r *http.Request) (err error, handled bool) {
 	opts := abisFinishParseApi(w, r)
+	outputHelpers.InitJsonWriterApi("abis", w, &opts.Globals)
 	// EXISTING_CODE
 	// EXISTING_CODE
-	return opts.AbisInternal()
+	err, handled = opts.AbisInternal()
+	outputHelpers.CloseJsonWriterIfNeededApi("abis", err, &opts.Globals)
+	return
```

Comment:
down into AbisInternal (and the associated routines under each command) and combine the four functions into two that use the fact that we know ApiMode()?

It lessens the cognitive load because, if I'm not mistaken, these routines do the same thing just slightly different.

Perhaps you didn't do this for a reason? Maybe the return values on error?

Seems cleaner to me.

Maybe something like `InitJsonWriter` whose first line is:

```
if globals.IsApiMode() {
    return InitJsonWriterApi(....)
}
```

If there's a reason why it's like this, I'm open to discussion.